### ISSUE TITLE
Switching interpreter to global in environment manager doesn't deactivate previously activated .venv

### ISSUE THREAD

**Issue by anthonykim1** [microsoft/vscode-python-environments#60]
Testing microsoft/vscode-python#24503

Video example: 

https://github.com/user-attachments/assets/a5c1d085-0b38-43b9-a51e-780546e7df4f

Having .venv activated and then switching the environment in the environment manager to non-venv such as python3.13, does not seem to change the terminal state. 

When I click the terminal icon next to my workspace name (after switching from venv to non-venv interpreter option), terminal is created with previously activated (now expected to be deactivated but wrongly persists as activated) venv. 

In additional to deactivate&activate issues(such as https://github.com/microsoft/vscode-python-environments/issues/49) that I have previously created, perhaps shell environment API will also be a key in solving this.

**Comment by karthiknadig** [comment]
@anthonykim1 Do you have the python extension env variable experiment on? This extension explicitly activates by sending a command to the terminal. If you have the python extension experiment on, then it can lead to activations because the variables were already set.

**Comment by karthiknadig** [comment]
When using this extension turn of both of these:
![Image](https://github.com/user-attachments/assets/70ef0fd2-4c21-4907-9077-db395b51d14a)

**Comment by anthonykim1** [comment]
Good point, I definitely had the terminal env var experiment on and probably one or both of the mentioned setting on as well. 
I should remember to turn off all three in future testing.  Probably helpful to note it as some of them are defaulted to true and would be easier to track down issues that are only specific to env manager.

**Comment by eleanorjboyd** [comment]
checked and this no longer happens
